cpu/cortex-m4_common/thread_arch.c

```diff
@@ -156,50 +160,54 @@ void thread_arch_yield(void)
     SCB->ICSR |= SCB_ICSR_PENDSVSET_Msk;
 }
 
-__attribute__((always_inline)) static __INLINE void context_save(void)
-{
-    /* {r0-r3,r12,LR,PC,xPSR} are saved automatically on exception entry */
-
-    /* save unsaved registers */
-    asm("mrs    r0, psp"            );      /* get stack pointer from user mode */
-    asm("stmdb  r0!,{r4-r11}"       );      /* save regs */
-    asm("stmdb  r0!,{lr}"           );      /* exception return value */
-/*  asm("vstmdb sp!, {s16-s31}"     ); */   /* TODO save FPU registers if needed */
-    asm("ldr    r1, =sched_active_thread" );      /* load address of current tcb */
-    asm("ldr    r1, [r1]"           );      /* dereference pdc */
-    asm("str    r0, [r1]"           );      /* write r0 to pdc->sp means current threads stack pointer */
-}
-
-__attribute__((always_inline)) static __INLINE void context_restore(void)
-{
-    asm("ldr    r0, =sched_active_thread" );      /* load address of current TCB */
-    asm("ldr    r0, [r0]"           );      /* dereference TCB */
-    asm("ldr    r1, [r0]"           );      /* load tcb->sp to register 1 */
-    asm("ldmia  r1!, {r0}"          );      /* restore exception return value from stack */
-/*  asm("pop    {s16-s31}"          ); */   /* TODO load FPU register if needed depends on r0 exret */
-    asm("ldmia  r1!, {r4-r11}"      );      /* restore other registers */
-    asm("msr    psp, r1"            );      /* restore PSP register (user mode SP)*/
-    asm("bx     r0"                 );      /* load exception return value to PC causes end of exception*/
-
-    /* {r0-r3,r12,LR,PC,xPSR} are restored automatically on exception return */
-}
-
 /**
  * @brief The svc is used for running the scheduler and scheduling a new task during start-up or
  *        after a thread has exited
  */
-__attribute__((naked)) void isr_svc(void)
-{
-    sched_run();
-    context_restore();
-}
+void isr_svc(void);
```

Comment:
Maybe I should add a normal (non-Doxygen) comment hinting that the definition (implementation) is found inside the naked function at the end?